packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -361,7 +361,7 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
           // because there is no way to check whether the default export is used
           (source.slice(expStart, start).includes('from') || isDynamicImport) &&
           // already has ?used query (by import.meta.glob)
-          !specifier.match(/\?used(&|$)/) &&
+          !specifier.match(/\?(used|raw|url)(&|$)/) &&
```

Comment:
I think we can use `SPECIAL_QUERY_RE` to check the special queries, and leave `!specifier.match(/\?used(&|$)/)` as is.

PS: I also tried to implement the idea I mentioned in discord last week. It's tricky since dynamic imports of css makes it harder to process it, which reminds me that _I think_ we haven't logged the warnings for dynamic imports of CSS yet.

Reply:
What warnings?

Reply:
We have some warnings at https://github.com/vitejs/vite/blob/a1019f80a5d5b6242d8fb0975994ce1ae6e78e94/packages/vite/src/node/plugins/importAnalysis.ts#L527-L531 and https://github.com/vitejs/vite/blob/a1019f80a5d5b6242d8fb0975994ce1ae6e78e94/packages/vite/src/node/plugins/importMetaGlob.ts#L330-L334 that discourages the use of default import of css, because that needs the `?used` hack. I don't think it's related to this PR now though, and we can go in with this quick fix for now.

Reply:
Alright. Done with `SPECIAL_QUERY_RE`

Reply:
> Alright. Done with `SPECIAL_QUERY_RE`

Speaking of dynamic import CSS, is there an alternative?

Reply:
> Speaking of dynamic import CSS, is there an alternative?

`import('./foo.css').then(mod => mod.default)` should now be replaced with `import('./foo.css?inline').then(mod => mod.default)`. Dynamic import without using default import (`import('./foo.css')`) is fine.